addons/gamification/models/res_users.py

```diff
@@ -69,18 +71,32 @@ def add_karma(self, karma):
         return True
 
     def _rank_changed(self):
-        if self.rank_id.karma_min > 0:
-            template = self.env.ref('gamification.mail_template_data_new_rank_reached', raise_if_not_found=False)
-            if template:
-                template.send_mail(self.id, force_send=True, notif_layout='mail.mail_notification_light')
+        """
+            Method that can be called on a batch of users with the same new rank
+        """
+        template = self.env.ref('gamification.mail_template_data_new_rank_reached', raise_if_not_found=False)
+        if template:
+            for u in self:
```

Comment:
`u` doesn't say anything, better variable name? ;)

Reply:
u: user, k: karma, r: rank

Reply:
self = res.users
so` for u in self`, it is implicit what is u => a user

It is not like if I create at the beginning of a long function a variable x and save something later into this variable.

def do_it():
    x = False
    ...
    [1000 lines of code]
    x = 3
    return True

Reply:
The time to write 'user', 'rank' or 'karma' instead of 'u', 'k', or 'r' is not nearly as close as thousands of people browsing through code and figuring out what it refers to, right?